crates/indexer/src/geyser/client.rs

```diff
@@ -84,12 +108,41 @@ impl Client {
         config_address: Pubkey,
         uri: String,
     ) -> Result<(), indexer_rabbitmq::Error> {
-        self.http
+        self.http_prod
             .store_config
             .write(http_indexer::StoreConfig {
                 config_address,
                 uri,
             })
             .await
     }
+
+    /// Dispatch a POST request to Dialect
+    ///
+    /// # Errors
+    /// This function fails if the underlying POST request results in an error.
+    pub async fn dispatch_dialect_offer_event(
+        &self,
+        bid_receipt_address: Pubkey,
+        metadata_address: Pubkey,
+    ) -> Result<()> {
+        let msg = DialectEvent {
+            event_type: DialectEventType::NftMakeOffer,
+            data: DialectEventData::DialectOfferEventData(DialectOfferEventData {
+                bid_receipt_address: bid_receipt_address.to_string(),
+                metadata_address: metadata_address.to_string(),
+            }),
+        };
+
+        self.http
+            .run(|h| {
+                h.post("https://holaplex.dialectapi.to/")
+                    .basic_auth("holaplex", Some(env::var("DIALECT_API_PW").unwrap()))
```

Comment:
This should be handled by clap, not read manually.  I can walk you through how to accomplish this.